src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.hpp

```diff
@@ -0,0 +1,130 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file ObstacleAvoidance.hpp
+ * This class is used to inject the setpoints of an obstacle avoidance system
+ * into the FlightTasks
+ *
+ * @author Martina Rivizzigno
+ */
+
+#pragma once
+
+#include <px4_defines.h>
+#include <px4_module_params.h>
+#include <commander/px4_custom_mode.h>
+#include <drivers/drv_hrt.h>
+
+#include <uORB/topics/position_controller_status.h>
+#include <uORB/topics/vehicle_command.h>
+#include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/vehicle_trajectory_waypoint.h>
+
+#include <matrix/matrix/math.hpp>
+
+#include <SubscriptionArray.hpp>
+
+class ObstacleAvoidance : public ModuleParams
+{
+public:
+	ObstacleAvoidance(ModuleParams *parent);
+	~ObstacleAvoidance();
+
+	bool initializeSubscriptions(SubscriptionArray &subscription_array);
+
+	/**
+	 * Inject setpoints from obstacle avoidance system into FLightTasks.
+	 * @param pos_sp, position setpoint
+	 * @param vel_sp, velocity setpoint
+	 * @param yaw_sp, yaw setpoint
+	 * @param yaw_speed_sp, yaw speed setpoint
+	 */
+	void prepareAvoidanceSetpoints(matrix::Vector3f &pos_sp, matrix::Vector3f &vel_sp, float &yaw_sp, float &yaw_speed_sp);
+
+	/**
+	 * Updates the desired waypoints to send to the obstacle avoidance system.
+	 * @param curr_wp, current position triplet
+	 * @param curr_yaw, current yaw triplet
+	 * @param curr_yawspeed, current yaw speed triplet
+	 * @param next_wp, next position triplet
+	 * @param next_yaw, next yaw triplet
+	 * @param next_yawspeed, next yaw speed triplet
+	 */
+	void updateAvoidanceDesiredWaypoints(const matrix::Vector3f &curr_wp, const float curr_yaw, const float curr_yawspeed,
```

Comment:
Might I suggest to stress it's informing an external component opposed to directly influencing the flight?
```suggestion
	void publishAvoidanceDesiredWaypoints(const matrix::Vector3f &curr_wp, const float curr_yaw, const float curr_yawspeed,
```

Reply:
There is already `_publishAvoidanceDesiredWaypoint` that does only the publishing. I left it as `updateAvoidanceDesiredWaypoints` but I added the comment about the effect of the message on te flight

Reply:
ok, it's the same for the position controller here https://github.com/PX4/Firmware/blob/d2a0c857edfc64d1b97ebd8965b7d1cfb0e182da/src/modules/mc_pos_control/mc_pos_control_main.cpp#L745-L746 and it was probably just me misunderstanding update since it does update a member of the class 👍